File: 70-ClimbingStairs/70-ClimbingStairs.py
# Last updated: 8/12/2026, 11:32:39 AM
class Solution:

    
    def climbStairs(self, n: int) -> int:
        
        if n<=0:
            return n

        dp = [0] * (n + 1)

        if dp[n]!=0:
            return dp[n]
        dp[0]=1
        dp[1]=1
        for i in range(2,n+1):
            dp[i]=dp[i-1]+dp[i-2]
        
        return dp[n]

File: 70-ClimbingStairs/70-ClimbingStairs.java
// Last updated: 8/12/2026, 11:32:34 AM
class Solution {
    static int[] dp = new int[100];
    
    public int climbStairs(int n) {
        if(dp[0]!=1){
            dp[0]=1;
            dp[1]=1;
        }
        if(n<=1){
            return n;
        }

        if(dp[n]!=0){
            return dp[n];
        }
        
        for(int i=2;i<=n;i++){
            dp[i]=dp[i-1]+dp[i-2];
        }

        return dp[n];
    }
}